packages/nimble-components/src/table/index.ts

```diff
@@ -11,7 +22,117 @@ declare global {
 /**
  * A nimble-styled table.
  */
-export class Table extends FoundationElement {}
+export class Table<
+    TData extends TableRowData = TableRowData
+> extends FoundationElement {
+    @observable
+    public data: TData[] = [];
+
+    // TODO: Temporarily expose the table data as arrays of strings.
+    @observable
+    public tableData: string[][] = [];
+
+    // TODO: Temporarily expose the column headers as a string array.
+    @observable
+    public columnHeaders: string[] = [];
+
+    private readonly table: TanStackTable<TData>;
+    private options: TanStackTableOptionsResolved<TData>;
+    private readonly tableInitialized: boolean = false;
+
+    public constructor() {
+        super();
+        const initialData = this.data;
+        this.options = {
+            get data(): TData[] {
+                return initialData;
+            },
```

Comment:
This seems a little odd, the parameter is initialized with an empty array object and the constructor is the first thing that runs so initialData is always pointing to the empty array and that means this getter is always returning that empty array

Is all that intended? Is the accessor specifically needed for some reason, we can't just do `data: this.data`?

Reply:
Ah, I had pulled this from the prototype branch without the proper consideration.

I think what makes the most sense is to simplify this to:
```ts
this.options = {
    data: [],
    ...
};
```

And, then `this.options` gets updated with any data changes through the `dataChanged` handler.